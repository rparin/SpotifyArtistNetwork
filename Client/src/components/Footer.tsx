import { FOOTER_PRIVACY, FOOTER_SEARCH_TEXT, CR } from "@/constants";
import { ExternalLink } from "lucide-react";
import { cn } from "@/lib/utils";

export default function Footer(props: { className?: string }) {
  return (
    <div
      className={cn(
        "bg-black text-white mx-auto flex flex-col items-center gap-2 py-2 w-full z-50",
        props.className
      )}>
      {/* <section className="flex gap-5 text-sm">
        <a
          href=""
          target="_blank"
          rel="noopener noreferrer"
          className="hover:underline underline-offset-2">
          <p>{FOOTER_SEARCH_TEXT}</p>
        </a>
        <a
          href=""
          target="_blank"
          rel="noopener noreferrer"
          className="hover:underline underline-offset-2">
          <p>{FOOTER_PRIVACY}</p>
        </a>
      </section> */}
      <section className="flex">
        <p className="text-white/30 text-xs">{CR}</p>
        <a
          href="https://rparin.github.io"
          aria-label="External link to rparin portfolio website"
          target="_blank"
          rel="noopener noreferrer">
          <ExternalLink className="opacity-40 w-auto h-[.7rem]" />
        </a>
      </section>
    </div>
  );
}
