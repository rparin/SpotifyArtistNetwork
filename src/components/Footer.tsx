import { CR, SOURCE_CODE } from "@/constants/AppConstants";
import { ExternalLink } from "lucide-react";
import cn from "@/utils/cn";

export default function Footer(props: { className?: string }) {
  return (
    <footer
      className={cn(
        "z-50 mx-auto flex w-full flex-row items-center justify-center gap-1 bg-black py-3 text-white",
        props.className
      )}>
      <p className="text-xs text-white/80">{CR}</p>
      <a
        href={SOURCE_CODE}
        aria-label="External link to source code"
        target="_blank"
        rel="noopener noreferrer">
        <ExternalLink className="mb-2 h-[.7rem] w-auto opacity-70" />
      </a>
    </footer>
  );
}
