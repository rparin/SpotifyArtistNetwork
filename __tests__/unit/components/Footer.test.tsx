import { render } from "@testing-library/react";
import { axe, toHaveNoViolations } from "jest-axe";
import Footer from "@/components/Footer";

test("it renders component unchanged", () => {
  const { container } = render(<Footer />);
  expect(container).toMatchSnapshot();
});

describe("testing Footer Accessibility", () => {
  expect.extend(toHaveNoViolations);

  test("it should have no accessibility violations", async () => {
    const { container } = render(<Footer />);
    const results = await axe(container);
    expect(results).toHaveNoViolations();
  });
});
