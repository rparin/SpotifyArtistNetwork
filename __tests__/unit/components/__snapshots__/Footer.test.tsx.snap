// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it renders component unchanged 1`] = `
<div>
  <footer
    class="z-50 mx-auto flex w-full flex-row items-center justify-center gap-1 bg-black py-3 text-white"
  >
    <p
      class="text-xs text-white/80"
    >
      ©️ 2024. Developed by Ralph Parin.
    </p>
    <a
      aria-label="External link to source code"
      href="https://github.com/rparin/SpotifyArtistNetwork/tree/Express"
      rel="noopener noreferrer"
      target="_blank"
    >
      <svg
        class="lucide lucide-external-link mb-2 h-[.7rem] w-auto opacity-70"
        fill="none"
        height="24"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M15 3h6v6"
        />
        <path
          d="M10 14 21 3"
        />
        <path
          d="M18 13v6a2 2 0 0 1-2 2H5a2 2 0 0 1-2-2V8a2 2 0 0 1 2-2h6"
        />
      </svg>
    </a>
  </footer>
</div>
`;
